Issue 612 aws ec2 using gui - closes #612 (#628)
bob-crutchley/courseware: topics/aws/modules/aws-ec2-gui/README.md

<!--TOC_END-->
## Overview
An EC2 instance is a virtual server in Amazon’s Elastic Compute Cloud (EC2) for running applications on the Amazon Web Services (AWS) infrastructure.

AWS is a comprehensive, evolving cloud computing platform; EC2 is a service that allows business subscribers to run application programs in the computing environment. The EC2 can serve as a practically unlimited set of virtual machines. 

You will use the AWS Dashboard to create virtual machine.

## Tutorial

1. Navigate to the AWS Console and sign in [here][aws-signin]

2. Search for EC2 under the services dropdown menu, and click on EC2, shown in the screenshot.

![ec2-search][ec2-search]

3. On the left panel, navigate to **Instances**

![instances-nav][instances-nav]

4. Click on the **Launch Instance** button to get started

![launch-instance-button][launch-instance-button]

5. You will be shown a large number of AMIs. Scroll down and select the **Ubuntu 18.04 LTS** AMI

![ubuntu-18-ami][ubuntu-18-ami]

6. You will be directed to a list of tiers which you can select. By default, **t2.micro** should be selected, if not, select **t2.micro** and continue to configuring details.

> 7. Here, you will be configuring your EC2 instance regarding what network your EC2 instance should sit in and so on. Leave everything as their default settings and continue to **Add Storage**. Details on this section can be found under the [More information](#more-information)

Reviewer: ```suggestion
7. In the next window, you will be configuring your EC2 instance regarding what network your EC2 instance should be deployed to. Leave everything as their default settings and continue to **Add Storage**. Details on this section can be found under the [More information](#more-information)
```

![instance-configuration][instance-configuration]

> 8. Here you will define how much additional storage you want to attach to your EC2 instance. The default storage will be used to install the OS. This will be the most minimal storage the OS needs to run. The second storage will be attached to your EC2 instance but you will need to mount the additional volume to a directory in your OS.

Reviewer: ```suggestion
8. In the next window, you will define how much additional storage you want to attach to your EC2 instance. The default storage will be used to install the OS. This will be the most minimal storage the OS needs to run. The second storage will be attached to your EC2 instance but you will need to mount the additional volume to a directory in your OS.
```

If you use a **Windows** server then this will be a 30GB storage for your OS to run.

Click **Next: Add Tags** and continue

![storage-configuration][storage-configuration]

> 9. You can make a key-value pair tags for your instance(s), so you can identify what your instance is being used for or provide a date of creation which can be used for serverless services. Leave it empty and conitue onto **Next: Configure Security Group** 

Reviewer: ```suggestion
9. You can make a key-value pair tags for your instance(s), so you can identify what your instance is being used for or provide a date of creation which can be used for serverless services. Leave it empty and continue onto **Next: Configure Security Group** 
```

10. Security group is similar to firewall rules. As you will define the ports you want to open and the source IP Address you want that port to be open to. Most of the time you will only configure the inbound rules and leave the outbound rules to its default setting. 

By deault, the inbound rule has port 22 open to the whole world, meaning anyone in the world can SSH into the EC2 instance provided they have the credentials. Leave it as its default setting and continue to **Review and Launch** 

![security-group-configuration][security-group-configuration]

> 11. Once youre happy with all the configuration, click **Launch**. You will be prompted to select an existing key or create a new key. Select **Create a new key pair**, and provide a name for your key pair. Then download the key or you will not be able to continue.

Reviewer: ```suggestion
11. Once youre happy with all the configuration, click **Launch**. You will be prompted to select an existing key or create a new key. Select **Create a new key pair**, and provide a name for your key pair. 

**NOTE:** Then download the key or you will not be able to continue.
```

![download-key-pair][download-key-pair]

Click **Launch Instances** 

12. Click **View Instances** to view all of your running instances. And you will see a new instance being created. You will have to wait until this creation is completed and all checks have passed.

> 13. Lets SSH into the EC2 Instance. Naviagte to where your key has been downloaded, usually in the Download folders. and use the following command:

Reviewer: ```suggestion
13. Lets SSH into the EC2 Instance. Navigate to where your key has been downloaded to, usually in the Download folder, and execute the following command:
```

```bash

chmod 400 <Your pem key>
ssh -i <Your pem key> ubuntu@<EC2 instance IP Address>

```

If you are using a **Windows** machine, you will not need to run the **chmod** command. If you're using a **Windows** machine, download **git bash** for windows

The IP Address can be found by navigating to all your running [instances][list-of-instances], select the running instance you want to SSH into. Under the **Description** tab, you will find **IPv4 Public IP**.

![instance-description-ip][instance-description-ip]

> You can also use the **Connect** button, this will provide the command to SSH into your server which you just follow the instruction.

Reviewer: ```suggestion
You can also use the **Connect** button, this will provide the command to SSH into your server which you just follow the instruction. You can copy the example SSH command and execute it in the terminal to connect to the VM, make sure your working directory is the one where the key is located.
```

![ssh-connect-button][ssh-connect-button]

> 14. To ensure you are not using up all your free-tier usage limit, you will need to **termina** your EC2 instance. 

Reviewer: ```suggestion
14. To ensure you are not using up all your free-tier usage limit, you will need to **terminate** your EC2 instance. 
```

    1. Select the instance you want to terminate
    2. Click on the **Action** button
    3. Go to **Instance State** then **Terminate**

> 

Reviewer: ```suggestion

**Note:** the terminated instance will appear in the list of instances for some time until it will be removed, this is normal behavior

```

![instance-terminate][instance-terminate]

## More information

### Number of Instances

#### Overview
You define the number of instances which you want to create based on the the AMI and tier you previously selected.

### Network

#### Overview
You define the **Virtual Private Cloud** which you want your EC2 instance(s) to sit within. This could mean that you could create instances that are not available to the rest of the world.

### Subnet

#### Overview
You define the specific location which your EC2 instance(s) are going to sit within the Network. 

### Auto-assign IP Address

#### Overview

> By default this should be set to **Use subnet setting (Enable)**, if not, then change to this option. This will automatically assign an IP address for internet access. 

Reviewer: ```suggestion
By default this should be set to **Use subnet setting (Enable)**, if not, then change to this option. This will automatically assign an IP address for internet access. Without a public IP you wouldn't be able to reach the VM from your computer.
```
